src/settings/nursery.rs

```diff
@@ -0,0 +1,17 @@
+use crate::registry::Rule;
+
+/// Rules for which a reason is returned here are disabled by default unless they are explicitly selected.
+/// So for example `one-blank-line-before-class` (D203) will not be selected by "D" or "ALL" since it's listed here.
+/// The returned reason is documented in the autogenerated README as well as reported by the explain command.
+pub fn nursery_reason(rule: &Rule) -> Option<&'static str> {
+    match rule {
+        Rule::OneBlankLineBeforeClass => Some("Conflicts with PEP 257."),
```

Comment:
Here's where I'm torn: assume that no user _ever_ enabled this rule. Why support it? Why maintain the code? The tests? Etc.

Reply:
I am alright with dropping these rules completely. However you don't want to break configs that currently ignore these rules and since the `RuleCodePrefix` serde impls are currently automatically derived, I think the easiest way to achieve that would be to introduce a no-op rule and map the legacy codes to that no-op violation that simply never gets triggered ... however since we don't yet have the many-to-many mapping we'd need to assign a rule code to the no-op rule ... which feels wrong since it's very much an implementation detail. So I'd rather have us wait with the dropping of these rules until after the many-to-many mapping has been implemented.

And I also think it would be nice to have some deprecation period with ruff still supporting the rules but printing a warning that we are planning on dropping them with a link to a GitHub issue about that, so that in case these rules are in fact used by some users they can voice their opinions about it.

Reply:
This is reasonable. I'd like to get these rules out of `--select D` and `--select ALL` as soon as we can. And the approach you've proposed here does achieve that (as long as we preserve retaining `D212` when the appropriate conventions are set, as per the discussion above).

I do think it's strange to put these rules in `nursery` rather than mark them as explicitly deprecated (and use the strategy you've implemented here to handle those deprecations), since we know we want to remove them. My preference would be to leave `S101` and `D212` for now, and use the strategy here for `D203` and `D213`, but mark them as deprecated rather than in nursery. To me, nursery suggests that there's a path to stabilizing these rules, but the intent here really is to discourage and stop supporting them.

Reply:
I am not convinced that keeping these rules around actually is much of a maintenance burden ... once the conflict / endless iteration issue has been fixed by disabling them by default they really shouldn't be such a hassle anymore ... but if you want to remove them that's very much your call.